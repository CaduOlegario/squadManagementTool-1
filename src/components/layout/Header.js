import React from 'react';
import {AppBar, Toolbar, Grid, Avatar, Typography} from "@material-ui/core";
import StringUtils from "../../utils/StringUtils";

const Header = ({username}) => {
    return (
        <AppBar position="static">
            <Toolbar>
                <Grid container spacing={1} alignItems="center" wrap="nowrap">
                    <Grid item>
                        <Avatar alt="Venturus Logo" src="/images/venturus_logo.png"/>
                    </Grid>
                    <Grid item>
                        <Typography variant="h6" noWrap>
                            Squad Management Tool
                        </Typography>
                    </Grid>
                    <Grid item xs={12}></Grid>
                    <Grid item>
                        <Typography variant="h6" noWrap>
                            {username}
                        </Typography>
                    </Grid>
                    <Grid item>
                        <Avatar>{StringUtils.getNameInitials(username)}</Avatar>
                    </Grid>
                </Grid>
            </Toolbar>
        </AppBar>
    )
};

export default Header;