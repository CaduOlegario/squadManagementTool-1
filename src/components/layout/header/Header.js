import React from 'react';
import {AppBar, Avatar, createMuiTheme, Grid, MuiThemeProvider, Toolbar, Typography} from "@material-ui/core";
import StringUtils from "utils/StringUtils";
import useTheme from "@material-ui/core/styles/useTheme";
import PropTypes from "prop-types";

const getMuiTheme = (theme) =>
    createMuiTheme({
        ...theme,
        overrides: {
            MuiAppBar: {
                colorPrimary: {
                    background: "linear-gradient(to right, " + theme.palette.primary.main + ", " + theme.palette.primary.dark + ")"
                }
            },
            MuiAvatar: {
                colorDefault: {
                    color: "black",
                    backgroundColor: "white"
                }
            }
        }
    });

/**
 * Header component that displays the application logo, name, the logged user and it's initials
 */
const Header = ({username}) => {
    return (
        <MuiThemeProvider theme={getMuiTheme(useTheme())}>
            <AppBar position="static">
                <Toolbar>
                    <Grid container spacing={1} alignItems="center" wrap="nowrap">
                        <Grid item>
                            <Avatar alt="Venturus Logo" src="/images/venturus_logo.png"/>
                        </Grid>
                        <Grid item>
                            <Typography variant="h6" noWrap>
                                Squad Management Tool
                            </Typography>
                        </Grid>
                        <Grid item xs={12}></Grid>
                        <Grid item>
                            <Typography variant="h6" noWrap>
                                {username}
                            </Typography>
                        </Grid>
                        <Grid item>
                            <Avatar>
                                <Typography variant="h6">
                                    {StringUtils.getNameInitials(username)}
                                </Typography>
                            </Avatar>
                        </Grid>
                    </Grid>
                </Toolbar>
            </AppBar>
        </MuiThemeProvider>
    )
};

Header.propTypes = {
    /** Name of the logged user */
    username: PropTypes.string.isRequired
};

export default Header;