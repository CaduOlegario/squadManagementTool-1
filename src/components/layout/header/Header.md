Header example:

```js
<Header username="Raul Seixas" />
```