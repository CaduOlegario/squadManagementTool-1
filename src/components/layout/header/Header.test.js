import React from 'react';
import renderer from 'react-test-renderer';
import Header from "./Header";

it('renders correctly', () => {
    const tree = renderer
        .create(<Header username="Test User Name"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});