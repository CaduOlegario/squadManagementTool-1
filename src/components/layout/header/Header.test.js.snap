// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle"
        >
          <img
            alt="Venturus Logo"
            className="MuiAvatar-img"
            src="/images/venturus_logo.png"
          />
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
        >
          Squad Management Tool
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      />
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
        >
          Test User Name
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
          >
            TU
          </h6>
        </div>
      </div>
    </div>
  </div>
</header>
`;
